### PR TITLE
[serializer] document DEEP_OBJECT_TO_POPULATE

### PR DESCRIPTION
**Pull request by dbu**
Document the new feature added in https://github.com/symfony/symfony/pull/30607

### CODE REVIEW

File: components/serializer.rst [hunk 1]
```diff
@@ -210,6 +210,16 @@ The serializer can also be used to update an existing object::
 
 This is a common need when working with an ORM.
 
+The OBJECT_TO_POPULATE is only used for the top level object. If that object
+is the root of a tree structure, all child elements that exist in the
+normalized data will be re-created with new instances.
+
+..versionadded:: 4.3
+
+    Symfony 4.3 introduces a new option AbstractObjectNormalizer::DEEP_OBJECT_TO_POPULATE.
```
Comment (xabbuh) [inline components/serializer.rst]:
```rst
``AbstractObjectNormalizer::DEEP_OBJECT_TO_POPULATE``
```

File: components/serializer.rst [hunk 2]
```diff
@@ -210,6 +210,16 @@ The serializer can also be used to update an existing object::
 
 This is a common need when working with an ORM.
 
+The OBJECT_TO_POPULATE is only used for the top level object. If that object
```
Comment (xabbuh) [inline components/serializer.rst]:
```rst
``DEEP_OBJECT_TO_POPULATE``
```

File: components/serializer.rst [hunk 3]
```diff
@@ -210,6 +210,16 @@ The serializer can also be used to update an existing object::
...
+is the root of a tree structure, all child elements that exist in the
+normalized data will be re-created with new instances.
+
+..versionadded:: 4.3
+
+    Symfony 4.3 introduces a new option AbstractObjectNormalizer::DEEP_OBJECT_TO_POPULATE.
+    When this flag is set to true, existing children of the root OBJECT_TO_POPULATE are
```
Comment (xabbuh) [inline components/serializer.rst]:
```rst
``DEEP_OBJECT_TO_POPULATE``
```

**Review by xabbuh — APPROVED** [review]
with minor comments

**Comment by wouterj** [comment]
Thank you @dbu! I've done a couple changes (mostly moving stuff out of the versionadded) in fed4d14a2 . Let me know if you disagree with anything.

**Comment by dbu** [comment]
ah, cool! the way you edited it is much better because we can remove the versionadded thing without rewriting any of the rest. i was afraid of people on older versions reading the doc and not noticing that its about a new feature, but if doc is written this way its better readable.

**Pull request merged**